compiler/tests-common/tests/org/jetbrains/kotlin/ir/AbstractIrTextTestCase.kt

```diff
@@ -164,7 +164,9 @@ abstract class AbstractIrTextTestCase : AbstractIrGeneratorTestCase() {
             private val elements = HashSet<IrElement>()
 
             override fun visitElement(element: IrElement) {
-                require(elements.add(element)) { "Non-unique element: ${element.render()}" }
+                require(elements.add(element) || element is IrConst<*>) {
```

Comment:
I've tracked down where this was added, and it's for checking if there is a duplicate array access (represented as `IrGetVariable`). I strongly believe having same const should not matter. Actually, "constant canonicalization" (so that the local function can avoid loading the same constants again) is well-known optimization.